### DISCUSSION TITLE
Sixel Preview?

### DISCUSSION THREAD

**Discussion by wzhchin** [junegunn/fzf#3479] [Show and tell]
In fzf 0.43, support for kitty's image protocol has been added. However, many people do not use kitty or terminal emulators that support the kitty image protocol. Should we consider supporting other protocols, such as sixel? I have already written a demo (inspired by lf-sixel), but I am facing some issues, such as the input line shifting up by one line in some layouts(My vocation is over, so I have no time working on this). So I wanted to ask if the community has any plans to implement this.

link: https://github.com/aeghn/fzf/tree/dev

preview: 
[preview.webm](https://github.com/junegunn/fzf/assets/51843252/8ed7159a-055a-4701-b022-3abf63eb46ee)

(the number lines is made for scroll test).

**Comment by junegunn** [comment]
I was hesitant to add support for a non-standard protocol but I decided to support Kitty image protocol, because

1. I'm always on tmux and it works on tmux. Does sixel work on tmux? If it doesn't work on tmux, I'll probably never use it, which would make me less motivated to maintain the feature.
2. It was trivial to implement it. All I had to do was pass through a chunk of text to the underlying terminal.
    * https://github.com/junegunn/fzf/commit/d8188fce7b7bea982e7f9050c35e488e49fb8fd0
    * Looking at your commit https://github.com/aeghn/fzf/commit/ca24ae9e4b1d65c6b03b9068e2d97e68b1b856a2, it's not as trivial as Kitty support.

**Reply by wzhchin** [reply]
Thanks for the reply, 

> Does sixel work on tmux?
Tmux didn't support sixel at the moment.

> it's not as trivial as Kitty support.

I have installed kitty, but image preview is not worked for me.

```sh
$ kitty --version
kitty 0.29.2 created by Kovid Goyal
```
![23-10-15_20 17 52](https://github.com/junegunn/fzf/assets/51843252/760c1ce2-ffbf-4058-8506-d72fac09885f)

And I made a little change based on 3666448ca649093f6f6b534b67e5a8b865d11a00, https://github.com/aeghn/fzf/commit/86555454c3c3da7cb6797cdfc4f46a8d119b3d67, then sixel image could show now(but there are need some logic for image detection, like image size and image retention cleaning).

**Reply by junegunn** [reply]
You probably need a newer version of kitty. If it doesn't work on the latest release, try installing the nightly build.

https://github.com/junegunn/fzf/issues/3228#issuecomment-1732888311


> And I made a little change based on [3666448](https://github.com/junegunn/fzf/commit/3666448ca649093f6f6b534b67e5a8b865d11a00), [aeghn@8655545](https://github.com/aeghn/fzf/commit/86555454c3c3da7cb6797cdfc4f46a8d119b3d67)

Looks promising.

**Comment by linwaytin** [comment]
According to https://www.arewesixelyet.com/, `tmux` is supported. Also, although it is not a standard, many terminal emulators support `sixel`.

**Comment by junegunn** [comment]
See https://github.com/junegunn/fzf/issues/2544
